samcli/commands/local/cli_common/options.py

```diff
@@ -5,6 +5,8 @@
 import click
 from samcli.commands._utils.options import template_click_option, docker_click_options, parameter_override_click_option
 
+from samcli.local.docker.manager import ContainerManager
```

Comment:
I find this import strange. I understand why but the Options shouldn't need to interact with a ContainerManger. Maybe we should place the `is_valid_container_name` somewhere else?

Reply:
It can go to utils or something, but for me, ContainerManager should be the only one responsible for the whole Docker management cycle. It will semantically be better, what do you think?

Reply:
To me validation isn't part of that management cycle, it more of a attribute of the container itself not the class that manages. 
@sanathkr or @TheSriram what are your thoughts here. I just want to make sure we keep the right abstractions is all.

Reply:
@jfuss, if we treat it like an attribute we can simply inline this method here.

Reply:
I more meant on the `Container` class instead of the `ContainerManger`. This isn't a huge deal for me, just something I noticed while reading through.

Reply:
I am ok with this abstraction you have. I think your reasoning is good. I think we just have a slightly different model for what is in the Docker management cycle. I am more worried about the use of `container_name` outside of debugging than this :)

Reply:
@jfuss, okay, then 😅 As for the container_name I do agree, that it is more about debugging, then something else. I will get back to PC, consider options and get back to you really nice catch.

Also, as related question, what do you think about determining whether `DebugContext` is true with `debug_port or debugger_path` as debugging port does not make sense for don't core debugging and customer would be forced to supply unused argument to get it working

Reply:
It might make sense to add one more abstraction to debugContext, with the transport. Since technically dotnet debugging can be done over ssh or docker. I'm not sure about other languages.

We still want to be able support ssh based debugging too. Thoughts?